csrc/device_lower/validation.cpp

```diff
@@ -753,7 +753,6 @@ class VectorizeValidator : public OptInDispatch {
     // Validate producer
     if (GpuLower::current()->hasIdModel()) {
       // No need to do replayPasC when using IdModel
-      NVF_ERROR(tv_def->input(0) == producer_tv);
```

Comment:
Unrelated, but this assertion was just there because I thought it would be always the case, which isn't true when `where` is directly vectorized.